registry/quilt_server/views.py

```diff
@@ -1556,21 +1570,26 @@ def profile():
     )
 
 @app.route('/api/payments/update_plan', methods=['POST'])
-@api()
+@api(enabled=HAVE_PAYMENTS)
 @as_json
 def payments_update_plan():
-    if not HAVE_PAYMENTS:
-        raise ApiException(requests.codes.not_found, "Payments not enabled")
-
     plan = request.values.get('plan')
     try:
         plan = PaymentPlan(plan)
```

Comment:
Actually, things like this are why `PaymentPlan` needs to contain all possible values.